core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/DatasourceServiceManagerGwtImpl.java

```diff
@@ -283,37 +284,47 @@ public void remove( IDatasourceInfo dsInfo, final Object xulCallback ) {
       removeURL = null;
     }
 
-    AuthenticatedGwtServiceUtil.invokeCommand( new IAuthenticatedGwtCommand<Boolean>() {
-      public void execute( final AsyncCallback<Boolean> callback ) {
-        RequestBuilder requestBuilder = new RequestBuilder( RequestBuilder.POST, removeURL );
-        try {
-          requestBuilder.sendRequest( null, new RequestCallback() {
-            @Override
-            public void onError( Request request, Throwable exception ) {
-              callback.onFailure( exception );
-            }
-
-            @Override
-            public void onResponseReceived( Request request, Response response ) {
-              callback.onSuccess( response.getStatusCode() == Response.SC_OK );
-            }
-
-          } );
-        } catch ( RequestException e ) {
-          XulServiceCallback<Boolean> responseCallback = (XulServiceCallback<Boolean>) xulCallback;
-          responseCallback.error( e.getLocalizedMessage(), e );
-        }
-      }
-    }, new AsyncCallback<Boolean>() {
+    CsrfUtil.getCsrfToken( removeURL, new AsyncCallback<JsCsrfToken>() {
+      public void onFailure( Throwable caught ) { }
```

Comment:
In this case, the previous code handled the failure case. Should we not do so as well?